### ISSUE TITLE
False positive with match_on_same_arms and macros

### ISSUE THREAD

**Issue by jugglerchris** [rust-lang/rust-clippy#1390]
Hi,

I have a trace macro which normally expands to nothing, but can be turned on by a Cargo feature.  I get `match_same_arms` warning on a case like this:

```rust
macro_rules! trace {
    ($e:expr) => ()
}

pub enum Foo { A, B, C }
use Foo::*;

pub fn test(x: Foo) {
    match x {
        A => { println!("something"); },
        B => { trace!("Got B"); },
        _ => {},
    }
}
```

which gives the following warning:

```
warning: this `match` has identical arm bodies, #[warn(match_same_arms)] on by default
  --> src/lib.rs:15:14
   |
15 |         _ => {},
   |              ^^
   |
note: same as this
  --> src/lib.rs:14:14
   |
14 |         B => { trace!("Got B"); },
   |              ^^^^^^^^^^^^^^^^^^^^
note: `B` has the same arm body as the `_` wildcard, consider removing it`
  --> src/lib.rs:14:14
   |
14 |         B => { trace!("Got B"); },
   |              ^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#match_same_arms
```

In some sense they expand to the same, but they're not the same to me the programmer; when I turn on a feature the tracing becomes more useful.

Playground link: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=929ad50b7c6d82e2df6bf3a8cbd7a944

**Comment by oli-obk** [comment]
yea... we do have some issues with `cfg`... As a hack, you could expand to calling `nop()`, which is defined as `#[inline(always)] fn nop() {}`.

**Comment by jugglerchris** [comment]
Thanks - that seems like a perfectly good workaround to me.

**Comment by phansch** [comment]
This doesn't trigger anything these days: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e181dfcd46e34ff50915d75e52789f78

**Comment by taiki-e** [comment]
@phansch: I don't think this also has been fixed yet: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=929ad50b7c6d82e2df6bf3a8cbd7a944

**Comment by phansch** [comment]
Oh, you're right, I forgot to actually enable the lint :facepalm: Thank you!
